pkg/graveler/graveler.go

```diff
@@ -3050,6 +3084,27 @@ func (g *Graveler) DiffUncommitted(ctx context.Context, repository *RepositoryRe
 			return nil, err
 		}
 	}
+	if branch.CompactedBaseMetaRangeID != "" {
+		if metaRangeID == "" {
+			compactedIterator, err := g.CommittedManager.List(ctx, repository.StorageNamespace, branch.CompactedBaseMetaRangeID)
+			if err != nil {
+				valueIterator.Close()
+				committedValueIterator.Close()
+				return nil, err
+			}
+			// if no meta range (no commit data) return the diff of staging + sealed + compacted from committed (empty iterator)
+			return NewUncommittedDiffIterator(ctx, committedValueIterator, NewCombinedIterator(valueIterator, compactedIterator)), nil
```

Comment:
If there's no commit data, how can you use `committedValueIterator`?

Reply:
All the data from the combined iterator will show as uncommitted